### PR TITLE
Fix toplevel transpile

### PR DESCRIPTION
**Pull request by aethos**

### CODE REVIEW

File: test/test.sh [hunk 1]
```diff
@@ -88,52 +87,43 @@ test_compiler() {
...
-    # compile program to test.ll, put any errors in the session
-    $oscar_compile $oscar_test_file 1> $cppfile 2> $errorout
-    $CXXCOMPILE -o $outfile $cppfile
 
-    echo "" > $testout
+    echo -e "Oscar Compiler Messages:" >> session_file
+    $oscar_compile $oscar_test_file 2> oscar_error_output
 
-    # if we had a error, then diff errors. Otherwise, run LLVM and diff outputs
-    if [ -s $errorout ]; then
```
Comment (aethos) [inline test/test.sh:103]: i know this reverts stuff here but it's necessary for compliance with toplevel

File: src/oscar.ml [hunk 2]
```diff
@@ -83,11 +101,26 @@ let _ =
...
+          | _ ->
+              let program = Transpile.c_program soprogram in
+              let file_stub = get_file_stub oscar in
+              let cpp_file = file_stub ^ ".cpp" in
+              let c_op = "-Wall -pedantic -fsanitize=address -std=c++1y -O2" in
+              let cxx_incls = "-I/usr/local/include/ " in
+              let cxx = (match action with
+                    Ast      -> ""
+                  | Sast     -> ""
+                  | Llvm_gen ->
```
Comment (Howon) [inline src/oscar.ml]: you can still clean this up

File: src/oscar.ml [hunk 3]
```diff
@@ -1,62 +1,80 @@
...
+    let (action, optimize, oscar) =
+    (if (arg_len < 2 || arg_len > 4) then
+      let _ = 
+          print_endline("Usage: ./oscar [-p|-s|-l|-c] [?-O] *.oscar") in
       exit 1;
     else if arg_len = 2 then
-      (Compile, true, open_in Sys.argv.(1))
-    else if arg_len = 3 then
+      (Compile, true, Sys.argv.(1))
+    else if Sys.argv.(2) = "-O" then
```
Comment (Howon) [inline src/oscar.ml]: why?

Comment (Howon) [inline src/oscar.ml]: can you make this branching cleaner

File: src/oscar.ml [hunk 4]
```diff
@@ -1,62 +1,80 @@
...
+                ("-s", Sast);     (* Prettyprint sast *)
+                ("-l", Llvm_gen); (* Generate cpp and llvm *)
+                ("-c", Compile);  (* Generate cpp and executable *)
+        ] Sys.argv.(1), false, Sys.argv.(2))
       with Not_found ->
         raise (Failure ("Invalid flag " ^ Sys.argv.(1)))
-    else raise (Failure ("Invalid flag " ^ Sys.argv.(2)))
-  ) in
-  let lexbuf = make_lexbuf oscar
+    else raise (Failure ("Invalid flag " ^ Sys.argv.(2)))) in
```
Comment (Howon) [inline src/oscar.ml:62]: i like the style in master better

**Review by Howon — APPROVED** [review]

**Pull request merged**